src/core_write.cpp

```diff
@@ -230,9 +230,28 @@ void TxToUniv(const CTransaction& tx, const uint256& hashBlock, bool include_add
             }
             in.pushKV("txinwitness", txinwitness);
         }
-        if (calculate_fee) {
-            const CTxOut& prev_txout = txundo->vprevout[i].out;
+        if (have_undo) {
+            const Coin& prev_coin = txundo->vprevout[i];
+            const CTxOut& prev_txout = prev_coin.out;
+
             amt_total_in += prev_txout.nValue;
+            switch (verbosity) {
+                case TxVerbosity::SHOW_TXID:
+                case TxVerbosity::SHOW_DETAILS:
+                    break;
+
+                case TxVerbosity::SHOW_DETAILS_AND_PREVOUT:
+                    UniValue o_script_pub_key(UniValue::VOBJ);
+                    ScriptPubKeyToUniv(prev_txout.scriptPubKey, o_script_pub_key, /* includeHex */ true, /* include_addresses */ true);
```

Comment:
Question: There is `include_addresses` parameter which is not used here. `true` is passed instead. Is that correct or should it look like:

```suggestion
                    ScriptPubKeyToUniv(prev_txout.scriptPubKey, o_script_pub_key, /* includeHex */ true, include_addresses);
```

It looks to me that `include_addresses = true` is deprecated actually.

Reply:
Since this is adding new output, I think it should probably just be a constant `false` here, actually.